v2/datastream-to-spanner/terraform/samples/all-infrastructure/README.md

```diff
@@ -0,0 +1,554 @@
+
+```shell
+Outputs:
+
+resource_ids = {
+  "dataflow_job" = "2024-06-14_03_01_00-3421054840094926119"
+  "datastream_source_connection_profile" = "source-mysql-thorough-wombat"
+  "datastream_stream" = "mysql-stream-thorough-wombat"
+  "datastream_target_connection_profile" = "target-gcs-thorough-wombat"
+  "gcs_bucket" = "live-migration-thorough-wombat"
+  "pubsub_subscription" = "live-migration-thorough-wombat-sub"
+  "pubsub_topic" = "live-migration-thorough-wombat"
+}
+resource_urls = {
+  "dataflow_job" = "https://console.cloud.google.com/dataflow/jobs/us-central1/2024-06-14_03_01_00-3421054840094926119?project=your-project-here"
+  "datastream_source_connection_profile" = "https://console.cloud.google.com/datastream/connection-profiles/locations/us-central1/instances/source-mysql-thorough-wombat?project=your-project-here"
+  "datastream_stream" = "https://console.cloud.google.com/datastream/streams/locations/us-central1/instances/mysql-stream-thorough-wombat?project=your-project-here"
+  "datastream_target_connection_profile" = "https://console.cloud.google.com/datastream/connection-profiles/locations/us-central1/instances/target-gcs-thorough-wombat?project=your-project-here"
+  "gcs_bucket" = "https://console.cloud.google.com/storage/browser/live-migration-thorough-wombat?project=your-project-here"
+  "pubsub_subscription" = "https://console.cloud.google.com/cloudpubsub/subscription/detail/live-migration-thorough-wombat-sub?project=your-project-here"
+  "pubsub_topic" = "https://console.cloud.google.com/cloudpubsub/topic/detail/live-migration-thorough-wombat?project=your-project-here"
+}
+```
+
+**Note:** Each of the jobs will have a random suffix added to it to prevent name
+collisions.
+
+### Cleanup
+
+Once the jobs have finished running, you can clean up by running -
+
+```shell
+terraform destroy --var-file=terraform_simple.tfvars
+```
+
+#### Note on GCS buckets
+
+The GCS bucket that is created will be cleaned up during `terraform destroy`.
+If you want to exclude the GCS bucket from deletion due to any reason, you
+can exclude it from the state file using `terraform state rm` command.
+
+## FAQ
+
+### Configuring to run using a VPC
+
+#### Datastream
+
+> **_NOTE:_** By default, **IP Whitelisting** based connectivity is assumed.
+
+There is a variable of the type below in `variables.tf` -
+
+```shell
+private_connectivity = optional(object({
+      private_connectivity_id = string
+      vpc_name                = string
+      range                   = string
+    }))
+```
+
+To enable private connectivity based access for Datastream, specify this
+configuration in your `*.tfvars`. For example -
+
+```shell
+ ...
+ mysql_databases = [
+    {
+      database = "lorem"
+    }
+  ]
+  private_connectivity = {
+    private_connectivity_id = "abc"
+    range                   = "xyz"
+    vpc_name                = "pqr"
+  }
+  ...
+```
+
+Note that `vpc_name` and `range` are mandatory and for the private connectivity
```

Comment:
Please add links to the GCP documentation for some of these

Reply:
Added